src/ResourceManagement/Batch/Microsoft.Azure.Management.V2.Batch/BatchAccountsImpl.cs

```diff
@@ -0,0 +1,113 @@
+/**
+* Copyright (c) Microsoft Corporation. All rights reserved.
+* Licensed under the MIT License. See License.txt in the project root for
+* license information.
+*/
+
+namespace Microsoft.Azure.Management.V2.Batch
+{
+
+    using Microsoft.Azure.Management.V2.Resource.Core.CollectionActions;
+    using Microsoft.Azure.Management.V2.Resource.Core;
+    using Microsoft.Azure.Management.V2.Storage;
+    using Microsoft.Azure.Management.V2.Batch.BatchAccount.Definition;
+    using Management.Batch.Models;
+    using Management.Batch;
+    using Resource;
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Implementation for BatchAccounts and its parent interfaces.
+    /// </summary>
+    public partial class BatchAccountsImpl :
+        GroupableResources<
+            IBatchAccount,
+            BatchAccountImpl,
+            AccountResourceInner,
+            IAccountOperations,
+            BatchManager>,
+        IBatchAccounts
+    {
+        private IStorageManager storageManager;
+
+        internal BatchAccountsImpl(IAccountOperations InnerCollection, BatchManager manager, IStorageManager storageManager)
+            : base(InnerCollection, manager)
+        {
+            this.storageManager = storageManager;
+        }
+
+        public void Delete(string id)
+        {
+            Delete(ResourceUtils.GroupFromResourceId(id), ResourceUtils.NameFromResourceId(id));
+        }
+
+        public void Delete(string groupName, string name)
+        {
+            this.InnerCollection.Delete(groupName, name);
+        }
+
+        protected override BatchAccountImpl WrapModel(string name)
+        {
+            AccountResourceInner inner = new AccountResourceInner();
+
+            return new BatchAccountImpl(
+                name,
+                inner,
+                InnerCollection,
+                MyManager, storageManager);
+        }
+
+        public PagedList<IBatchAccount> List()
+        {
+            return new ChildListFlattener<IResourceGroup, IBatchAccount>(MyManager.ResourceManager.ResourceGroups.List(), (IResourceGroup resourceGroup) =>
+            {
+                return ListByGroup(resourceGroup.Name);
+            }).Flatten();
+        }
+
+        public PagedList<IBatchAccount> ListByGroup(string resourceGroupName)
+        {
+            var pagedList = new PagedList<AccountResourceInner>(InnerCollection.ListByResourceGroup(resourceGroupName));
+            return WrapList(pagedList);
```

Comment:
Looks like SDK has `ListByResourceGroup` and `ListByResourceGroupNext`, that means above code will wrap only the first page .. to use the next link, code should be:

```C#
            IPage<AccountResourceInner> firstPage = InnerCollection.ListByResourceGroup(resourceGroupName);
            var pagedList = new PagedList<AccountResourceInner>(firstPage, (string nextPageLink) =>
            {
                return InnerCollection.ListByResourceGroupNext(nextPageLink);
            });
            return WrapList(pagedList);
```